src/v/model/transform.cc

```diff
@@ -190,6 +190,27 @@ void transform_metadata::serde_write(iobuf& out) const {
     serde::write(out, compression_mode);
 }
 
+void transform_metadata::serde_read(iobuf_parser& in, const serde::header& h) {
+    using serde::read_nested;
+
+    name = read_nested<decltype(name)>(in, h._bytes_left_limit);
+    input_topic = read_nested<decltype(input_topic)>(in, h._bytes_left_limit);
+    output_topics = read_nested<decltype(output_topics)>(
+      in, h._bytes_left_limit);
+    environment = read_nested<decltype(environment)>(in, h._bytes_left_limit);
+    uuid = read_nested<decltype(uuid)>(in, h._bytes_left_limit);
+    source_ptr = read_nested<decltype(source_ptr)>(in, h._bytes_left_limit);
+    offset_options = read_nested<decltype(offset_options)>(
```

Comment:
Don't we need a version check here? Only read this if version > 0, same for the version 2 fields only.

Or do these noop if there are no bytes left?

Reply:
If called, read functions expect that there are bytes left to read. E.g. here:
https://github.com/redpanda-data/redpanda/blob/10e3f4328799c81df78240af24a25ccedc02f619/src/v/serde/rw/scalar.h#L32-L39

That's why the default version for reading an `envelope` is okay if the `bytes_left` were depleted and skips reading the following fields (`compat_version` vs serialized `version` comparison made sure this is okay):
https://github.com/redpanda-data/redpanda/blob/10e3f4328799c81df78240af24a25ccedc02f619/src/v/serde/rw/envelope.h#L70-L74

Reply:
Ah, I completely misread this comment. Thanks @rockwotj